src/c++/perf_analyzer/test_request_rate_manager.cc

```diff
@@ -39,251 +39,295 @@ namespace triton { namespace perfanalyzer {
+    //
+    for (auto len : stats->sequence_status.seq_lengths) {
+      CHECK(len == doctest::Approx(params.sequence_length).epsilon(0.20));
+    }
+  }
+
+  std::vector<int64_t> GatherTimeBetweenRequests(
+      const std::vector<std::chrono::time_point<std::chrono::system_clock>>&
+          timestamps)
+  {
+    std::vector<int64_t> time_between_requests;
+
+    for (size_t i = 1; i < timestamps.size(); i++) {
+      auto diff = timestamps[i] - timestamps[i - 1];
+      std::chrono::nanoseconds diff_ns =
+          std::chrono::duration_cast<std::chrono::nanoseconds>(diff);
+      time_between_requests.push_back(diff_ns.count());
     }
+    return time_between_requests;
+  }
 
-    std::unique_ptr<LoadManager> CreateManager(PerfAnalyzerParameters params, bool is_sequence_model=false) {
+  double CalculateAverage(std::vector<int64_t> values)
+  {
+    double avg =
+        std::accumulate(values.begin(), values.end(), 0.0) / values.size();
+    return avg;
+  }
+
+  double CalculateVariance(std::vector<int64_t> values, double average)
+  {
+    double tmp = 0;
+    for (auto value : values) {
+      tmp += (value - average) * (value - average) / values.size();
+    }
+    double variance = sqrt(tmp);
+    return variance;
+  }
 
-      std::unique_ptr<LoadManager> manager;
-      std::shared_ptr<cb::ClientBackendFactory> factory = std::make_shared<cb::MockClientBackendFactory>(stats_);
-      std::shared_ptr<ModelParser> parser = std::make_shared<MockModelParser>(is_sequence_model);
+  std::unique_ptr<LoadManager> CreateManager(
+      PerfAnalyzerParameters params, bool is_sequence_model = false)
+  {
+    std::unique_ptr<LoadManager> manager;
+    std::shared_ptr<cb::ClientBackendFactory> factory =
+        std::make_shared<cb::MockClientBackendFactory>(stats_);
+    std::shared_ptr<ModelParser> parser =
+        std::make_shared<MockModelParser>(is_sequence_model);
 
-      RequestRateManager::Create(
+    RequestRateManager::Create(
         params.async, params.streaming, params.measurement_window_ms,
-        params.request_distribution, params.batch_size,
-        params.max_threads, params.num_of_sequences,
-        params.sequence_length, params.string_length,
+        params.request_distribution, params.batch_size, params.max_threads,
+        params.num_of_sequences, params.sequence_length, params.string_length,
         params.string_data, params.zero_input, params.user_data,
         params.shared_memory_type, params.output_shm_size,
-        params.start_sequence_id, params.sequence_id_range, parser,
-        factory, &manager);
-
-      return manager;
-    }        
+        params.start_sequence_id, params.sequence_id_range, parser, factory,
+        &manager);
 
+    return manager;
+  }
 };
 
 
 /// Check that the correct inference function calls
 /// are used given different param values for async and stream
 ///
-TEST_CASE("request_rate_infer_type") {
+TEST_CASE("request_rate_infer_type")
```

Comment:
Small suggestion: doctest `TEST_CASE()` macro accepts a string as the name in order to allow developer to be more descriptive in the code (which will show up to end user of unit tests if test fails). You could directly copy and paste your comment above the test into the test's name, or at the very least be more descriptive in the test name string.

Not needed for this PR, but something to keep in mind for future use of doctest 👍.

https://github.com/doctest/doctest/blob/master/doc/markdown/tutorial.md#what-did-we-do-here

Reply:
My goal was to make them easily filterable and selectable.  If I want to run a single testcase from the command line, I don't want to have to say `-tc="some huge paragraph that has to be spelled exactly the same"`

Reply:
Maybe both can suffice: `<identifier_name>: prose description`

Reply:
Also, I would argue that if unit tests are written to be observably instant, then there's not much value add for filtering.